mypy/semanal.py

```diff
@@ -1864,21 +1862,30 @@ def leave_class(self) -> None:
 
     def analyze_class_decorator(self, defn: ClassDef, decorator: Expression) -> None:
         decorator.accept(self)
+        self.analyze_class_decorator_common(defn, defn.info, decorator)
         if isinstance(decorator, RefExpr):
             if decorator.fullname in RUNTIME_PROTOCOL_DECOS:
                 if defn.info.is_protocol:
                     defn.info.runtime_protocol = True
                 else:
                     self.fail("@runtime_checkable can only be used with protocol classes", defn)
-            elif decorator.fullname in FINAL_DECORATOR_NAMES:
-                defn.info.is_final = True
-            elif refers_to_fullname(decorator, TYPE_CHECK_ONLY_NAMES):
-                defn.info.is_type_check_only = True
         elif isinstance(decorator, CallExpr) and refers_to_fullname(
             decorator.callee, DATACLASS_TRANSFORM_NAMES
         ):
             defn.info.dataclass_transform_spec = self.parse_dataclass_transform_spec(decorator)
 
+    def analyze_class_decorator_common(
+        self, defn: ClassDef, info: TypeInfo, decorator: Expression
+    ) -> None:
+        """Common part for all class decorators.
```

Comment:
```suggestion
        """Common hook for applying class decorators.
```

Reply:
Done! Thanks, except it is not a hook. By hooks we often mean things like `self.plugin.get_customize_class_mro_hook(defn.fullname)`